### PR TITLE
Make setters fluent

### PR DESCRIPTION
**Pull request by elkuku**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| Tickets       | 
| License       | MIT

I just replaced some charts in a personal project with ux-chartjs and it feels quite cool ;)

To reduce local variables i'd love to see fluent setters. 

wdyt?

### CODE REVIEW

File: src/Chartjs/Model/Chart.php [hunk 1]
```diff
@@ -38,19 +38,25 @@ public function __construct(string $type)
         $this->type = $type;
     }
 
-    public function setData(array $data)
+    public function setData(array $data): self
```
Comment (nicolas-grekas) [inline src/Chartjs/Model/Chart.php:41]: can you please add the `@return $this` annotation?
also can you please check if some other UX component could benefit from fluent setters?

Comment (elkuku) [reply src/Chartjs/Model/Chart.php:41]: > can you please add the `@return $this` annotation?
You mean the doc block? is this still required?

> also can you please check if some other UX component could benefit from fluent setters?
From what I can see the UX-Cropperjs package could also benefit from fluent setters. should I include the changes here or open a separate PR?

Comment (nicolas-grekas) [reply src/Chartjs/Model/Chart.php:44]: > You mean the doc block? is this still required?

yep, they are more precise than just `: self`

> should I include the changes here

yes please, in the same PR

Comment (elkuku) [reply src/Chartjs/Model/Chart.php:44]: Done ;)

**Review by nicolas-grekas — APPROVED** [review]

**Comment by tgalopin** [comment]
Thank you @elkuku.

**Pull request merged**